lib/active_model_serializers/adapter/json_api/resource_identifier.rb

```diff
@@ -20,11 +20,13 @@ def as_json
 
         def type_for(serializer)
           return serializer._type if serializer._type
-          if ActiveModelSerializers.config.jsonapi_resource_type == :singular
-            serializer.object.class.model_name.singular
-          else
-            serializer.object.class.model_name.plural
+          type = serializer.object.class.to_s.split('::')
+          type.map! { |t| t.underscore.downcase }
```

Comment:
@bf4 Even though it makes perfect logical sense to have this in `KeyTransform`, cause it is a utility module that transforms keys, would it be too much of a performance hit to add the logic for the separator?

At the very least, this logic could be added as a new method in the `KeyTransform` module.